sw/device/lib/base/csr.h

```diff
@@ -0,0 +1,2786 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_BASE_CSR_H_
+#define OPENTITAN_SW_DEVICE_LIB_BASE_CSR_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/stdasm.h"
+
+// Define function implementations (mocked or real).
+#ifdef MOCK_CSR
+
+uint32_t mock_csr_read(const char *name);
+#define CSR_READ_IMPL(name, addr) return mock_csr_read(#name);
+
+uint32_t mock_csr_write(const char *name, uint32_t value);
+#define CSR_WRITE_IMPL(name, addr) return mock_csr_write(#name, value);
+
+uint32_t mock_csr_set_bits(const char *name, uint32_t mask);
+#define CSR_SET_BITS_IMPL(name, addr) return mock_csr_set_bits(#name, mask);
+
+uint32_t mock_csr_clear_bits(const char *name, uint32_t mask);
+#define CSR_CLEAR_BITS_IMPL(name, addr) return mock_csr_clear_bits(#name, mask);
+
+#else  // MOCK_CSR
+
+#define CSR_READ_IMPL(name, addr)                  \
```

Comment:
Could we simply use `name` without `addr`?

Similar to the below - using the name `pmpaddrN` instead of `0x3B0`?
`asm volatile("csrr %0, pmpaddr" #region_id ";" : "=r"(*value));`

It makes it less likely to make a mistake in `.hjson` file used for the generator (otherwise you need to make sure that name matches the correct CSR address).

Reply:
Hmmmmm I guess the answer is:

> Also note that not all CSR names are supported by the assembler so we even with macros we will need to at least have some constants to help with using them.

I didn't realise that - do we currently use any of CSR calls that don't support the name?

Reply:
> I didn't realise that - do we currently use any of CSR calls that don't support the name?

At the very least `mseccfg` which is part of the draft ePMP spec and so doesn't have assembler support.